services/squareone/templates/ingress.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "squareone.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    {{- include "squareone.labels" . | nindent 4 }}
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    {{- with .Values.ingress.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
```

Comment:
I think what I'd do here is have a simple boolean in `values.yaml` that's `ingress.tls` that you can set to true or false, and if it's set to true, insert the correct TLS stanza for `.Values.global.host`, hard-coding the secret name that we currently set in the values files. That minimizes the amount of stuff you have to put into the per-environment values files.

If that sounds like a good idea, I'd also add the required annotation if `ingress.tls` is set to true, since we always use the same cert-issuer name, and then you can remove that from values as well.